src/components/features/header.tsx

```diff
@@ -0,0 +1,39 @@
+'use client';
+
+import Link from 'next/link';
+import { useState } from 'react';
+
+import { Header } from '../Header';
+import { PromoBanner } from '../PromoBanner';
+import { barbsershopData } from '../PromoBanner/PromoData';
+import { Sidebar } from './sidebar';
+
+export function HeaderPage(): JSX.Element {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <Header />
+      <PromoBanner barbershopCount={barbsershopData.barbershopCount} />
+      {isOpen ? <Sidebar isOpen={isOpen} setIsOpen={setIsOpen} /> : <div />}
+      <div className="display: inline-grid absolute top-96 mt-44 ml-12">
+        <button
+          onClick={() => setIsOpen(!isOpen)}
```

Comment:
please read this and fix accordingly https://www.delftstack.com/howto/react/react-setstate-prevstate/